globus_automate_client/client_helpers.py

```diff
@@ -0,0 +1,82 @@
+    """
+    A helper function to handle creating a properly authenticated ``ActionClient``
+    which can operate against *Globus Action Provider Interface* compliant Action
+    Providers. This helper will look for and store tokens on a local filesystem,
+    optionally triggering a log-in flow if the requested tokens are not found
+    locally. To create an ActionClient without referencing local storage or
+    triggering a login flow, instantiate an authorizer directly and use the
+    ``ActionClient.new_client`` classmethod.
+
+    Given the ``action_url`` for a specific ActionProvider, this function will
+    attempt to create a valid ``ActionClient`` for interacting with that
+    ActionProvider. If the ``action_scope`` is not provided, this function will
+    attempt to discover the ``action_scope`` by querying the target Action
+    Provider's introspection endpoint. If the Action Provider is not configured
+    to allow public, unauthenticated access to its introspection endpoint, the
+    ``action_scope`` will be non-discoverable and authentication will fail.
+
+    With the ``action_scope`` available, the function will search for a valid
+    token using the fair-research/native-login library. In the event that tokens
+    for the scope cannot be loaded, an interactive login will be triggered. Once
+    tokens have been loaded, an Authorizer is created and used to instantiate
+    the ``ActionClient`` which can be used for operations against that Action
+    Provider.
+
+    :param action_url: The URL address at which the target Action Provider
+        exists
+    :param action_scope: The target Action Provider's Globus Auth Scope used
+        for authenticating access to it
+    """
+    authorizer = get_cli_authorizer(action_url=action_url, action_scope=action_scope)
+    return ActionClient.new_client(action_url=action_url, authorizer=authorizer)
+
+
+def cli_authorizer_callback(**kwargs):
+    flow_url = kwargs["flow_url"]
+    flow_scope = kwargs["flow_scope"]
+    client_id = kwargs["client_id"]
+    return get_cli_authorizer(flow_url, flow_scope, client_id)
+
+
+def create_flows_client(
+    client_id: str, base_url: str = PROD_FLOWS_BASE_URL
+) -> FlowsClient:
+    """
+    A helper function to handle creating a properly authenticated
+    ``FlowsClient`` which can operate against the Globus Automate Flows service.
+    This function will attempt to load tokens for the MANAGE_FLOWS_SCOPE using
+    the fair-research/native-login library. In the event that tokens for the
+    scope cannot be loaded, an interactive login Flow will be triggered. Once
+    tokens have been loaded, an Authorizer is created and used to instantiate
+    the ``FlowsClient``. To create a FlowsClient without referencing local
+    storage or triggering a login flow, instantiate an authorizer directly and
+    use the ``FlowsClient.new_client`` classmethod.
+
+    :param client_id: The Globus ID to associate with this instance of the
```

Comment:
Is there a reason we have `client_id` here but not on the similar action client helper?

Reply:
Not really, but I'm not totally sure what `client_id` does in either scenario. Is it just used for identifying the application? Should we care if the CLI ID spills over and is used at the SDK?

Reply:
The client id is used to identify the native client requesting the grant from Auth. Auth will display the name of that client when it requests consent. I think this is ok here as I see some difference between action and flows client.

But, in the bit of context I see here, it is hard to think about, so I'm gonna look again at doing a review of the most recent commit and see what I see with more context.